tests/std/tests/P0896R4_ranges_alg_uninitialized_copy_n/test.cpp

```diff
@@ -0,0 +1,142 @@
+    static constexpr int expected_input[]  = {13, 55, 12345};
+
+    template <ranges::input_range Read, ranges::forward_range Write>
+    static void call() {
+        using ranges::uninitialized_copy_n, ranges::uninitialized_copy_n_result, ranges::equal, ranges::equal_to,
+            ranges::iterator_t;
+
+        int_wrapper input[3] = {13, 55, 12345};
+        Read wrapped_input{input};
+        holder<int_wrapper, 3> mem;
+        Write wrapped_output{mem.as_span()};
+
+        int_wrapper::clear_counts();
+        const same_as<uninitialized_copy_n_result<iterator_t<Read>, iterator_t<Write>>> auto result =
+            uninitialized_copy_n(wrapped_input.begin(), 3, wrapped_output.begin(), wrapped_output.end());
+        assert(int_wrapper::constructions == 3);
+        assert(int_wrapper::destructions == 0);
+        assert(result.in == wrapped_input.end());
+        assert(result.out == wrapped_output.end());
+        assert(equal(wrapped_output, expected_output, equal_to{}, &int_wrapper::val));
+        assert(equal(input, expected_input, equal_to{}, &int_wrapper::val));
+        not_ranges_destroy(wrapped_output);
+        assert(int_wrapper::constructions == 3);
+        assert(int_wrapper::destructions == 3);
+    }
+};
+
+struct throwing_test {
+    static constexpr int expected_input[] = {13, 55, int_wrapper::magic_throwing_val, 12345};
+
+    template <ranges::input_range Read, ranges::forward_range Write>
+    static void call() {
+        int_wrapper input[] = {13, 55, int_wrapper::magic_throwing_val, 12345};
+        Read wrapped_input{input};
+        holder<int_wrapper, 4> mem;
+        Write wrapped_output{mem.as_span()};
+
+        int_wrapper::clear_counts();
+        try {
+            (void) ranges::uninitialized_copy_n(wrapped_input.begin(), 4, wrapped_output.begin(), wrapped_output.end());
+            assert(false);
+        } catch (int i) {
+            assert(i == int_wrapper::magic_throwing_val);
+        } catch (...) {
+            assert(false);
+        }
+        assert(int_wrapper::constructions == 2);
+        assert(int_wrapper::destructions == 2);
+    }
+};
+
+template <test::ProxyRef IsProxy>
+using test_input  = test::range<test::input, int_wrapper, test::Sized::no, test::CanDifference::no, test::Common::no,
+    test::CanCompare::yes, IsProxy>;
+using test_output = test::range<test::fwd, int_wrapper, test::Sized::no, test::CanDifference::no, test::Common::no,
+    test::CanCompare::yes, test::ProxyRef::no>;
+
+int main() {
+    // The algorithm is oblivious to non-required category, size, difference. It _is_ sensitive to proxyness in that it
+    // requires non-proxy references for the input range.
+
+    instantiator::call<test_input<test::ProxyRef::no>, test_output>();
+    instantiator::call<test_input<test::ProxyRef::yes>, test_output>();
+    throwing_test::call<test_input<test::ProxyRef::no>, test_output>();
+    throwing_test::call<test_input<test::ProxyRef::yes>, test_output>();

```

Comment:
This comment doesn't seem to match the code, which is varying input proxyness.

Reply:
Yeah it should be output